File: opt_tax_recur/calibrations/BGP.jl
#=

Author: Shunsuke Hori

=#

module BGP
using LS
function baseline(;
    beta = 0.9,
    psi = 0.69,
    Pi = 0.5 *ones(2,2),
    G = [0.1,0.2],
    Theta = ones(2),
    transfers = false)
    #derivatives of utiltiy function
    U(c,n) = log(c) + psi*log(1-n)
    Uc(c,n) = 1./c
    Ucc(c,n) = -c.^(-2.0)
    Un(c,n) = -psi./(1.0-n)
    Unn(c,n) = -psi./(1.0-n).^2.0
    n_less_than_one = true
    return LS.Para(beta, Pi, G, Theta, transfers,
                    U, Uc, Ucc, Un, Unn, n_less_than_one)
end

#Model 1
M1 = baseline()

#Model 2

M2 = baseline()
M2.G = [0.15]
M2.Pi = ones(1,1)
M2.Theta = [1.0]
end

File: opt_tax_recur/calibrations/BGP.py
# -*- coding: utf-8 -*-
"""
Created on Wed Feb 18 15:32:07 2015

@author: dgevans
"""
import numpy as np

class baseline_BGP:
    beta = 0.9
    psi = 0.69

    Pi = 0.5 *np.ones((2,2))

    G = np.array([0.1,0.2])

    Theta = np.ones(2)

    transfers = False

    #derivatives of utiltiy function
    def U(self,c,n):
        return np.log(c) + self.psi*np.log(1-n)

    def Uc(self,c,n):
        return 1./c

    def Ucc(self,c,n):
        return -c**(-2)

    def Un(self,c,n):
        return -self.psi/(1-n)

    def Unn(self,c,n):
        return -self.psi/(1-n)**2


#Model 1
M1_BGP = baseline_BGP()

#Model 2

M2_BGP = baseline_BGP()
M2_BGP.G = np.array([0.15])
M2_BGP.Pi = np.ones((1,1))
M2_BGP.Theta = np.ones(1)

#Model 3 with time varying

M_time_example_BGP = baseline_BGP()

M_time_example_BGP.Pi = np.array([[0., 1., 0.,   0.,  0., 0.],
                                  [0., 0., 1.,   0.,  0., 0.],
                                  [0., 0., 0., 0.5, 0.5,  0.],
                                  [0., 0., 0., 0.,   0.,  1.],
                                  [0., 0., 0., 0.,   0.,  1.],
                                  [0., 0., 0., 0.,   0.,  1.]])

M_time_example_BGP.G = np.array([0.1, 0.1, 0.1, 0.2, 0.1, 0.1])
M_time_example_BGP.Theta = np.ones(6) # Theta can in principle be random
